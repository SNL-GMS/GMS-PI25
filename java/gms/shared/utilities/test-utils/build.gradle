plugins {
    id('java-library')
}

dependencies {
    implementation project(':java-utilities')

    implementation libs.com.fasterxml.jackson.core.jackson.databind
    implementation libs.com.fasterxml.jackson.datatype.jackson.datatype.jdk8
    implementation libs.com.fasterxml.jackson.datatype.jackson.datatype.jsr310
    implementation libs.com.fasterxml.jackson.datatype.jackson.datatype.guava
    implementation libs.com.fasterxml.jackson.dataformat.jackson.dataformat.yaml
    implementation libs.org.assertj.assertj.core
    implementation libs.org.junit.jupiter.junit.jupiter.base
    implementation libs.org.msgpack.jackson.dataformat.msgpack
}
