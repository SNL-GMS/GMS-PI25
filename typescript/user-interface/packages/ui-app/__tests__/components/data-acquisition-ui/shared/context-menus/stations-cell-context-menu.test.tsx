import { render } from '@testing-library/react';
import React from 'react';
import { act } from 'react-dom/test-utils';

import type {
  StationSohContextMenuItemProps,
  StationSohContextMenuProps
} from '../../../../../src/ts/components/data-acquisition-ui/shared/context-menus/stations-cell-context-menu';
import {
  acknowledgeOnClick,
  DisabledStationSohContextMenu,
  getStationAcknowledgementMenuText,
  StationSohContextMenu
} from '../../../../../src/ts/components/data-acquisition-ui/shared/context-menus/stations-cell-context-menu';

const mockShowContextMenu = jest.fn();
const mockHideContextMenu = jest.fn();

jest.mock('@blueprintjs/popover2', () => {
  const actual = jest.requireActual('@blueprintjs/popover2');
  return {
    ...actual,
    showContextMenu: () => {
      mockShowContextMenu();
    },
    hideContextMenu: () => {
      mockHideContextMenu();
    }
  };
});

describe('Station cell context menu', () => {
  const stationSohContextMenuProps: StationSohContextMenuProps = {
    stationNames: ['test', 'test2'],
    acknowledgeCallback: jest.fn()
  };

  const stationSohContextMenu = render(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <StationSohContextMenu {...stationSohContextMenuProps} />
  );
  const disabledStationSohContextMenu = render(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <DisabledStationSohContextMenu {...stationSohContextMenuProps} />
  );

  it('StationSohContextMenu can be created', () => {
    expect(stationSohContextMenu.baseElement).toMatchSnapshot();
  });

  it('DisabledStationSohContextMenu can be created', () => {
    expect(disabledStationSohContextMenu.baseElement).toMatchSnapshot();
  });

  it('getStationAcknowledgementMenuText to work as expected', () => {
    const stationNames = ['test', 'test1'];
    const withComment = false;
    const expectedResult = 'Acknowledge 2 stations';
    const result = getStationAcknowledgementMenuText(stationNames, withComment);
    expect(result).toEqual(expectedResult);
  });

  it('acknowledgeOnClick to work as expected', async () => {
    const stationSohContextMenuItemProps: StationSohContextMenuItemProps = {
      ...stationSohContextMenuProps,
      disabled: false
    };
    const mouseEvent: any = {
      preventDefault: jest.fn(),
      clientX: 5,
      clientY: 5
    };
    // call acknowledge and wait for React lifecycle to finish so it can be deferred.
    await act(async () => {
      acknowledgeOnClick(mouseEvent, stationSohContextMenuItemProps);
      const waitDurationMs = 200;
      // eslint-disable-next-line no-promise-executor-return
      await new Promise(resolve => setTimeout(resolve, waitDurationMs));
    });
    expect(mockShowContextMenu).toHaveBeenCalled();
  });
});
