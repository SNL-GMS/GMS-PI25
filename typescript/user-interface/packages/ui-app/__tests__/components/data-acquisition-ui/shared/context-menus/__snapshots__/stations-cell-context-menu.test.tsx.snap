// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Station cell context menu DisabledStationSohContextMenu can be created 1`] = `<body />`;

exports[`Station cell context menu StationSohContextMenu can be created 1`] = `
<body>
  <div>
    <ul
      class="bp4-menu"
      role="menu"
    >
      <li
        class=""
        role="none"
      >
        <a
          class="bp4-menu-item bp4-popover2-dismiss acknowledge-soh-status"
          data-cy="acknowledge-without-comment"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            Acknowledge 2 stations
          </div>
        </a>
      </li>
      <li
        class=""
        role="none"
      >
        <a
          class="bp4-menu-item bp4-popover2-dismiss acknowledge-soh-status"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            Acknowledge 2 stations with comment...
          </div>
        </a>
      </li>
    </ul>
  </div>
  <div>
    <ul
      class="bp4-menu"
      role="menu"
    >
      <li
        class=""
        role="none"
      >
        <a
          class="bp4-menu-item bp4-disabled acknowledge-soh-status"
          data-cy="acknowledge-without-comment"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            Acknowledge 2 stations
          </div>
        </a>
      </li>
      <li
        class=""
        role="none"
      >
        <a
          class="bp4-menu-item bp4-disabled acknowledge-soh-status"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            Acknowledge 2 stations with comment...
          </div>
        </a>
      </li>
    </ul>
  </div>
</body>
`;
